test/org/sagebionetworks/bridge/play/controllers/AuthenticationControllerMockTest.java

```diff
@@ -96,7 +106,48 @@ public void before() {
         when(studyService.getStudy(TEST_STUDY_ID_STRING)).thenReturn(study);
         controller.setStudyService(studyService);
     }
-
+    
+    @Test
+    public void requestEmailSignIn() throws Exception {
+        mockPlayContextWithJson(TestUtils.createJson("{'study':'study-key','email':'email@email.com'}"));
+        
+        Result result = controller.requestEmailSignIn();
+        assertResult(result, 202, "Email sent.");
+     
+        verify(authenticationService).requestEmailSignIn(signInCaptor.capture());
+        assertEquals("study-key", signInCaptor.getValue().getStudyId());
+        assertEquals("email@email.com", signInCaptor.getValue().getEmail());
+    }
+    
+    @Test
+    public void emailSignIn() throws Exception {
+        StudyParticipant participant = new StudyParticipant.Builder().build();
+        mockPlayContextWithJson(TestUtils.createJson("{'study':'study-key','email':'email@email.com','token':'ABC'}"));
+        doReturn(participant).when(userSession).getParticipant();
+        study.setIdentifier("study-test");
+        doReturn(study).when(studyService).getStudy("study-test");
+        doReturn(true).when(userSession).isAuthenticated();
+        doReturn(userSession).when(authenticationService).emailSignIn(any(CriteriaContext.class), any(SignIn.class));
+        
+        Result result = controller.emailSignIn();
+        assertEquals(200, result.status());
+        JsonNode node = BridgeObjectMapper.get().readTree(Helpers.contentAsString(result));
+        assertTrue(node.get("authenticated").asBoolean());
```

Comment:
I didn't now that. I always mean booleanValue(), not asBoolean()... changed this.